src/Dofs/sparsity_pattern.jl

```diff
@@ -96,21 +100,78 @@ function _coupling_to_local_dof_coupling(dh::DofHandler, coupling::AbstractMatri
     return outs
 end
 
-function _create_sparsity_pattern(dh::AbstractDofHandler, ch#=::Union{ConstraintHandler, Nothing}=#, sym::Bool, keep_constrained::Bool, coupling::Union{AbstractMatrix{Bool},Nothing})
+"""
+    cross_element_coupling!(dh::DofHandler, topology::ExclusiveTopology, sym::Bool, keep_constrained::Bool, couplings::Union{AbstractVector{<:AbstractMatrix{Bool}},Nothing}, cnt::Int, I::Vector{Int}, J::Vector{Int})
+
+Mutates `I, J` to account for cross-element coupling in fields with discontinuous interpolations.
+Returns the updated value of `cnt`.
+
+Used internally for sparsity patterns with discontinuous interpolations.
+"""
+function cross_element_coupling!(dh::DofHandler, topology::ExclusiveTopology, sym::Bool, keep_constrained::Bool, couplings::Union{AbstractVector{<:AbstractMatrix{Bool}},Nothing}, cnt::Int, I::Vector{Int}, J::Vector{Int})
+    for (sdh_idx, sdh) in pairs(dh.subdofhandlers)
+        # Buffering interpolation information for type stability/better allocations
+        ip_infos = InterpolationInfo[]
+        nbasefunctions = Int[]
+        for ip in sdh.field_interpolations
+            ip_info = InterpolationInfo(ip)
+            push!(ip_infos, ip_info)
+            push!(nbasefunctions, getnbasefunctions(ip))
+        end
+        isnothing(couplings) || (coupling_sdh = couplings[sdh_idx])
+        for (cell_field_i, cell_field) in enumerate(sdh.field_names)
+            fii = ip_infos[cell_field_i]
+            # Couple element for discontinuous interpolations only
+            fii.is_discontinuous || continue
+            for cell_idx in eachindex(getcells(dh.grid))
+                # Making sure current field is defined for current cell
+                cell_field ∈ dh.subdofhandlers[dh.cell_to_subdofhandler[cell_idx]].field_names || continue
+                dofrange1 = dof_range(sdh, cell_field)
+                cell_dofs = celldofsview(dh, cell_idx)
+                cell_field_dofs = @view cell_dofs[dofrange1]
```

Comment:
You can probably use CellCaches here (instead of the view):
```
cc1 = CellCache(dh)
cc2 = CellCache(dh)

reinit!(cc1, cell_idx)
reinit!(cc2, neighbor_cell)
```
etc